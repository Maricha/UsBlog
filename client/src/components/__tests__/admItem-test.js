import React from 'react';
import renderer from 'react-test-renderer';
import AdmItem from '../admItem';
import { BrowserRouter } from "react-router-dom";
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { createHttpLink } from 'apollo-link-http'
import { InMemoryCache } from 'apollo-cache-inmemory';

describe('AdmItem Component', () => {
  it('renders correctly', () => {
    const post = {
      id: 1,
      title: "title"
    }

    const client = new ApolloClient({
      link: createHttpLink({ uri: 'http://localhost:3000' }),
      cache: new InMemoryCache()
    })

    const tree = renderer.create(
      <BrowserRouter>
        <ApolloProvider client={client}>
          <AdmItem post={post} />
        </ApolloProvider>
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});