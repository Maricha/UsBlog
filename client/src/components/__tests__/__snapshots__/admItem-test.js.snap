// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdmItem Component renders correctly 1`] = `
<div
  className="MuiGrid-container-5 Component-itemWrapper-1"
>
  <div
    className="MuiGrid-item-6 MuiGrid-grid-xs-8-41 MuiGrid-grid-md-10-71"
  >
    <a
      className="Component-itemLink-2"
      href="/admin/post/1/edit"
      onClick={[Function]}
    >
      <p>
        1
         
        title
      </p>
    </a>
  </div>
  <div
    className="MuiGrid-item-6 MuiGrid-grid-xs-2-35 MuiGrid-grid-md-1-62"
  >
    <div
      className="DeleteButton-wrapper-103"
    >
      <button
        className="DeleteButton-buttonWrapper-102"
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-6 MuiGrid-grid-xs-2-35 MuiGrid-grid-md-1-62 Component-editWrapper-3"
  >
    <a
      className="Component-editButton-4"
      href="/admin/post/1/edit"
      onClick={[Function]}
    >
      Edycja
    </a>
  </div>
</div>
`;
